cmass/diagnostics/summ.py

```diff
@@ -72,32 +165,33 @@ def halo_summ(source_path, L, N, h, z, threads=16, from_scratch=True):
                 hpos = f[a]['pos'][...]
                 hvel = f[a]['vel'][...]
                 hmass = f[a]['mass'][...]
-
-                # measure halo Pk in comoving space
-                delta = MA(hpos, L, N, MAS='NGP')
-                k, Pk = calcPk(delta, L, MAS='NGP', threads=threads)
-
-                # measure halo Pk in redshift space
-                delta = MAz(hpos, hvel, L, N, h, z, MAS='NGP')
-                kz, Pkz = calcPk(delta, L, MAS='NGP', threads=threads)
-
+                # Ensure all halos inside box
+                m = np.all((hpos >= 0) & (hpos < L), axis=1)
+                hpos = hpos[m]
+                hvel = hvel[m]
+                hmass = hmass[m]
+                # Get summaries in comoving space
+                group = o.create_group(a)
+                box_catalogue = get_box_catalogue(pos=hpos, z=z, L=L, N=N)
+                for summ in summaries:
+                    store_summary(box_catalogue, group, summ, L, N, threads, use_rsd=False,)
+
+                # Get summaries in redshift space
+                zbox_catalogue = get_box_catalogue_rsd(pos=hpos, vel=hvel, h=h, z=z, axis=2, L=L, N=N)
```

Comment:
This gives an error for me
```
 File "/home/x-dbartlett/ltu-cmass/cmass/diagnostics/summ.py", line 60, in get_box_catalogue_rsd
    pos = get_redshift_space_pos(pos=pos, vel=vel, z=z, h=h, axis=axis, L=L,)
  File "/home/x-dbartlett/ltu-cmass/cmass/diagnostics/tools.py", line 8, in get_redshift_space_pos
    RSL.pos_redshift_space(pos, vel, L, h*100, z, axis)
  File "library/redshift_space_library/redshift_space_library.pyx", line 29, in redshift_space_library.redshift_space_library.pos_redshift_space
ValueError: Buffer dtype mismatch, expected 'float' but got 'double'
```
Please ensure correct type is passed to the relevant function

Reply:
This has now been fixed by ensuring `pos` and `vel` are of dtype `np.float32`.